File: java_src/DH3DLibrary/src/jp/spec/darkhorse2/dh3d/base/VertexShader.java
/**
 * 
 */
package jp.spec.darkhorse2.dh3d.base;

import webgl.*;

/**
 * @author Norihiro
 *
 */
public class VertexShader {
	private WebGLShader _shader;
	private WebGLRenderingContex _gl;
	private String _name;
	private String _program;
	
	public void VertexShader(WebGLRenderingContex gl){
		
	}
	public void bufferDynamicVertexData(DH3DObject dhObject){
		
	}
	public String getName(){
		return this._name;
	}
	public WebGLShader getShader(){
		return this._shader;
	}
	public void setAttribPointer(){
		
	}
	public void setMaterialData(Material material){
		
	}
	public void bindAttribute(WebGLProgram programObject){
		
	}
	public void bindAttribute2(WebGLProgram programObject){
		
	}
	public void setLightData(Light light){
		
	}
	public WebGLFloatArray getVertexData(DH3DObject dhObject){
		return null;
	}
	public WebGLFloatArray getDynamicVertexData(DH3DObject dhObject){
		return null;
	}
}

File: src/js/base/VertexShader.js
'use strict'

/**
 * VertexShader class
 * @access public
 */
export default class VertexShader {
  /**
   * constructor
   * @access public
   * @param {WebGLRenderingContext} gl -
   * @constructor
   */
  constructor(gl) {
    this._gl = null
    this._shader = null

    if(!gl)
      return

    this._shader = gl.createShader(gl.VERTEX_SHADER)
    gl.shaderSource(this._shader, this._program)
    gl.compileShader(this._shader)
    if(!gl.getShaderParameter(this._shader, gl.COMPILE_STATUS)){
      // alert(gl.getShaderInfoLog(this._shader)) // FIXME: DEBUG
      const info = gl.getShaderInfoLog(this._shader)
      console.log(info)
      throw new Error('vertex shader compile error: ' + info)
    }
    this._gl = gl
  }

  get _name() {
    return 'VertexShader'
  }

  getName() {
    return this._name
  }

  get _program() {
    return null
  }

  getShader() {
    return this._shader
  }

  bindAttribute(programObject) {
  }

  bindAttribute2(programObject) {
  }

  bufferDynamicVertexData(dhObject) {
  }

  setAttribPointer() {
  }

  setLightData(light) {
  }

  setMaterialData(material) {
  }

  getVertexData(dhObject) {
  }

  getDynamicVertexData(dhObject) {
  }
}

